File: src/main/java/vip/yazilim/libs/springvip/exception/DatabaseException.java
package vip.yazilim.libs.springvip.exception;

/**
 * Fired when a database exception occurred
 *
 * @author Emre Sen, 26.06.2019
 * @contact maemresen@yazilim.vip
 */
public class DatabaseException extends RuntimeException {
    private static final long serialVersionUID = 5862962376829113206L;

    protected <E> DatabaseException(Class<E> entityClass, String type, String description, Exception exception) {
        super(entityClass.getSimpleName() + " :: " + type + " Error :: " + description, exception);
    }
}

File: src/main/kotlin/vip/yazilim/libs/springvip/util/generic/service/exception/DatabaseException.kt
package vip.yazilim.libs.springvip.util.generic.service.exception

import kotlin.reflect.KClass

/**
 * Fired when a database exception occurred
 *
 * @author Emre Sen (maemresen@yazilim.vip), 26.06.2019
 */
open class DatabaseException(entityClass: KClass<*>, type: String, description: String, exception: Exception)
    : RuntimeException(entityClass.simpleName + "::" + type + " Error::" + description, exception)